src/main/java/uk/gov/laa/ccms/caab/mapper/AssessmentMapper.java

```diff
@@ -0,0 +1,497 @@
+            proceeding.getLevelOfService().getId()),
+        toAttribute(MATTER_TYPE,
+            proceeding.getMatterType().getId()),
+        toAttribute(NEW_OR_EXISTING,
+            ProceedingUtil.getNewOrExisting(proceeding)),
+        toAttribute(PROCEEDING_ID,
+            ProceedingUtil.getOpaInstanceMappingId(proceeding)),
+        toAttribute(PROCEEDING_NAME,
+            proceeding.getProceedingType().getId()),
+        toAttribute(PROCEEDING_ORDER_TYPE,
+            proceeding.getTypeOfOrder().getId()),
+        toAttribute(REQUESTED_SCOPE,
+            ProceedingUtil.getRequestedScopeForAssessmentInput(proceeding)),
+        toAttribute(SCOPE_LIMIT_IS_DEFAULT,
+            ProceedingUtil.isScopeLimitDefault(proceeding)));
+  }
+
+  @Mapping(target = "id", ignore = true)
+  @Mapping(target = "name", source = "proceeding",
+      qualifiedByName = "getProceedingOpaInstanceMappingId")
+  @Mapping(target = "attributes", source = "proceeding",
+      qualifiedByName = "proceedingToAttributeList")
+  @Mapping(target = "prepopulated", constant = "true")
+  @Mapping(target = "relations", ignore = true)
+  @Mapping(target = "completed", ignore = true)
+  AssessmentEntityDetail toAssessmentEntityDetail(ProceedingDetail proceeding);
+
+  @Mapping(target = "id", ignore = true)
+  @Mapping(target = "name", source = "opponentContext.opponent",
+      qualifiedByName = "getOpponentOpaInstanceMappingId")
+  @Mapping(target = "attributes", source = "opponentContext",
+      qualifiedByName = "opponentToAttributeList")
+  @Mapping(target = "prepopulated", constant = "true")
+  @Mapping(target = "relations", ignore = true)
+  @Mapping(target = "completed", ignore = true)
+  AssessmentEntityDetail toAssessmentEntityDetail(AssessmentOpponentMappingContext opponentContext);
+
+  /**
+   * Converts a list of opponent mapping contexts to a list of {@link AssessmentEntityDetail}
+   * objects.
+   *
+   * @param opponentContextList the list of opponent mapping contexts
+   * @return a list of assessment entity details, or {@code null} if the input list is null
+   */
+  default List<AssessmentEntityDetail> toAssessmentEntityDetailListOpponent(
+      final List<AssessmentOpponentMappingContext> opponentContextList) {
+
+    if (opponentContextList == null) {
+      return null;
+    }
+
+    final List<AssessmentEntityDetail> list = new ArrayList<>(opponentContextList.size());
+
+    for (final AssessmentOpponentMappingContext opponentContext : opponentContextList) {
+      list.add(toAssessmentEntityDetail(opponentContext));
+    }
+
+    return list;
+  }
+
+  /**
+   * Converts opponent data to a list of {@link AssessmentAttributeDetail} objects.
+   *
+   * @param opponentContext the opponent mapping context containing opponent details
+   * @return a list of assessment attribute details for the opponent
+   */
+  @Named("opponentToAttributeList")
+  default List<AssessmentAttributeDetail> opponentToAttributeList(
```

Comment:
Please can these be compared to the files:

com.legalservices.laa.ccms.config.MappingConfig In PUI

see - public CcmsInstancePrepopulator foreachOpponent() {